Week two

no0dly opened the pull request:
### Домашнее задание номер два.
- Добавлен AuthContext
- Добавлены значения контекста для логина логаута и isLogged флага
- подключен конкест к нужным компонентам чере useContext
- написанны простейшие тесты для компонентов на проверку рендерится ли компонент без ошибок
- добавлена проверка типов пропсов.
- стилизованы хедер, логин, регистрация с помощью material-ui

продолжаю работать над тестами
![dz2](https://user-images.githubusercontent.com/8776095/93837209-1cd14180-fc3a-11ea-8aa8-aa067c070a3e.gif)

no0dly committed:
feat: added map

Added map

no0dly committed:
feat: styled header

Styled header with material ui

no0dly committed:
feat: styled form for registration page

styled form for registration page

no0dly committed:
feat: added auth consumer

Added auth context

no0dly committed:
feat: styled Login form

Styled login form and added header styles to showup on map page

no0dly committed:
refactor(authcontext): replaced logic related to auth contex to a new file

replaced logic related to auth contex to a new file

no0dly committed:
feat: added should exist tests

added should exist component tests

no0dly committed:
feat: added auth context test

added auth context test

no0dly committed:
feat: added test for header

Added test for header navigation

no0dly committed:
feat(app.test): added test for initial load

Danil00524 commented on src/App.js:
```diff
@@ -5,27 +5,33 @@ import Login from "./Login";
+    setPage(pageName);
   };
 
+  const isLoginPages = page === "LOGIN" || page === "REGISTRATION";
```
Через CAPS обычно пишут константы. Как вариант можешь за пределами компонента объявить константу pages как объект, а там уже будут капсом написаны эти названия. 
pages.LOGIN

no0dly replied:
Поправил

Danil00524 commented on src/AuthContext.js:
```diff
@@ -0,0 +1,41 @@
+  const login = (email, password) => {
+    console.log("Login is process", email, password);
+    setAuth(true);
+    console.log("Login is done");
```
Это бессмысленные консоли.

no0dly replied:
Поправил

Danil00524 commented on src/Login.js:
```diff
@@ -1,76 +1,109 @@
+                    type="submit"
+                    variant="contained"
+                    color="primary"
+                    // onClick={() => onPageChange("MAP")}
```
Удаляем закомментированный код.

no0dly replied:
Поправил

Danil00524 commented on src/Registration.js:
```diff
@@ -17,88 +31,93 @@ function Registration({ onPageChange }) {
+                  <Button
+                    variant="contained"
+                    color="primary"
+                    onClick={() => onPageChange("MAP")}
```
Такие обработчики вешаем на тэг form.

no0dly replied:
Поправил

no0dly committed:
refactor: review comments fix

Removed console.logs, replaced on click function to on submit registration component, added
constants file
